src/tools/rbd/action/Import.cc

```diff
@@ -188,19 +188,22 @@ static int do_import(librbd::RBD &rbd, librados::IoCtx& io_ctx,
   }
   r = throttle->wait_for_ret();
   if (r < 0) {
-    goto done;
+    goto err;
   }
 
   if (from_stdin) {
     r = image.resize(image_pos);
     if (r < 0) {
       std::cerr << "rbd: final image resize failed" << std::endl;
-      goto done;
+      goto err;
     }
   }
 
   r = image.close();
 
+err:
+  if (r < 0)
+    rbd.remove(io_ctx, imgname);
```

Comment:
I think the image would still be open in the goto case -- so this remove would fail.